modules/plymouth/nixos.nix

```
@@ -50,13 +50,7 @@ in {
       type = with types; either path package;
       defaultText = literalMD "NixOS logo";
 
-      # Considering that Flake inputs are currently unable to fetch individual
-      # files, the SVG file is fetched with `pkgs.fetchurl` to avoid downloading
-      # the entire repository for a single SVG file.
-      default = pkgs.fetchurl {
-        url = "https://raw.githubusercontent.com/NixOS/nixos-artwork/f84c13adae08e860a7c3f76ab3a9bef916d276cc/logo/nix-snowflake-colours.svg";
-        sha256 = "pHYa+d5f6MAaY8xWd3lDjhagS+nvwDL3w7zSsQyqH7A=";
-      };
+      default = "${pkgs.nixos-icons}/share/icons/hicolor/256x256/apps/nix-snowflake.png";
```

We should assert the validity of the `default` value:

```suggestion
      default = let
        path = "${pkgs.nixos-icons}/share/icons/hicolor/256x256/apps/nix-snowflake.png";
      in
        lib.throwIfNot
        (builtins.pathExists path)
        "No such file or directory: ${path}"
        path
```

This ensures that upstream file path changes are caught by our CI.

---

Resolved via https://github.com/danth/stylix/pull/424/commits/123c2a38bb872b4c962b163e9edbe254fb639f1b.

---

Does this introduce an import-from-derivation?

---

> Does this introduce an import-from-derivation?

`throwIfNot` is defined at https://github.com/NixOS/nixpkgs/blob/nixpkgs-24.05-darwin/lib/trivial.nix#L757-L797 as:

```nix
throwIfNot = cond: msg: if cond then x: x else throw msg;
```

Apparently, it does:

```console
$ nix repl '<nixpkgs>' --no-allow-import-from-derivation
nix-repl> let path = "${pkgs.nixos-icons}/share/icons/hicolor/256x256/apps"; in lib.throwIfNot (builtins.pathExists path) "No such file or directory: ${path}" path
error:
       … while evaluating a branch condition

         at /nix/store/k9pv2y1kax9c2bli9r8xncdzfnx3n9zw-nixos-23.05.4065.3b79cc4bcd9c/nixos/lib/trivial.nix:386:27:

          385|   */
          386|   throwIfNot = cond: msg: if cond then x: x else throw msg;
             |                           ^
          387|

       … while calling the 'pathExists' builtin

         at «string»:1:87:

            1| let path = "${pkgs.nixos-icons}/share/icons/hicolor/256x256/apps"; in lib.throwIfNot (builtins.pathExists path) "No such file or directory: ${path}" path
             |                                                                                       ^

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: cannot build '/nix/store/g1yk9gfngfrgbkpllqbvsmx0m4yxj552-nixos-icons-2021-02-24.drv^out' during evaluation because the option 'allow-import-from-derivation' is disabled
```

---

The current IFD can be avoided by setting a scheme manually, and directly within the user's flake, rather than importing from `pkgs.base16-schemes` for example.

IMO, we shouldn't introduce an unavoidable IFD just for a check which isn't strictly necessary.

#379 may provide an alternative way to prevent unexpected breakage.

---

> IMO, we shouldn't introduce an unavoidable IFD just for a check which isn't strictly necessary.

Agreed. https://github.com/danth/stylix/pull/424/commits/44048c2c3ae529fce014353cf2d1d44b75852803 removes the IFD again.

> #379 may provide an alternative way to prevent unexpected breakage.

True.